### PR TITLE
Devise views creation

### LINKED ISSUE #9: Password recovery

### LINKED ISSUE #10: Mailer configuration

### LINKED ISSUE #26: Sign up / sign in for users

### LINKED ISSUE #39: Registration confirmation

### PR DESCRIPTION
**Pull request by ghost**
Created user sign up, sign in and sign out processes.
Created account validation and password recovery processes.

Configured mailer for local and heroku using gmail account

closes #26 
closes #9 
closes #10 
closes #39

### CODE REVIEW

File: app/views/layouts/_header.html.slim [hunk 1]
```diff
@@ -5,19 +5,25 @@
...
+                = "Sign out"
+          - else
+            li
+              = link_to new_user_session_path, data: {toggle: 'tab'} do
+                span
+                  = "Sign In"
+            li
+              = link_to new_user_registration_path, data: {toggle: 'tab'} do
+                span
+                  = "Sign Up"
```
Comment (hav3loc) [inline app/views/layouts/_header.html.slim]: This way this is not working, I need your help @bpietraga

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,3 +1,21 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  before_action :authenticate_user!, :except => [:index]
```
Comment (hav3loc) [inline app/controllers/application_controller.rb]: We need to decide whether we do this in every controller or just here @maciek1995

**Comment by hav3loc** [comment]
In order to send emails one needs to create file `.env` in their project root directory and supply `GMAIL_USERNAME` and `GMAIL_PASSWORD` variables.

**Comment by bpietraga** [comment]
+1

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,7 @@
 class UsersController < ApplicationController
+
```
Comment (gotar) [inline app/controllers/users_controller.rb]: drop empty line

**Comment by gotar** [comment]
+1 ok you can merge

**Pull request merged**